src/Parallel/Algorithm.hpp

```diff
@@ -409,6 +540,66 @@ class AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>
     return number_of_actions;
   }
 
+  // Invoke the action's static `apply` method. The if constexprs are for
+  // handling the cases where the `apply` method returns a tuple of one, two,
+  // three, or four elements, in order:
+  // 1. a DataBox
+  // 2. a bool determining whether or not to terminate (and move
+  // to the next phase)
+  // 3. an unsigned integer corresponding to which action in the current phase's
+  //    algorithm to execute next.
+  // 4. a std::vector<metavariables::Phase> of phases to request to jump to
```

Comment:
It's annoying me that an action provides the requests as a vector, which we first convert to an unordered_set, and then into a set. Can you convince me this is reasonable? I'd otherwise suggest making the actions provide their requests as an unordered_set, too.

Reply:
I'm happy to have the actions send `unordered_set`s